mockkubeapiserver/patchresource.go

```diff
@@ -124,6 +128,13 @@ func (req *patchResource) Run(ctx context.Context, s *MockKubeAPIServer) error {
 		klog.Infof("skipping write, object not changed")
 		return req.writeResponse(existingObj)
 	} else {
+		bumpGeneration := !reflect.DeepEqual(existingObj.Object["spec"], updated.Object["spec"])
+		if bumpGeneration {
```

Comment:
nit: Intent might be clearer expressed here if we move the negation to the conditional:
```suggestion
		specIsSame := reflect.DeepEqual(existingObj.Object["spec"], updated.Object["spec"])
		if !specIsSame {
```

Reply:
Great suggestion, thank you.  Applying!